### ISSUE TITLE
[Feature Request] Ability to select a OpenAI model

### ISSUE THREAD

**Issue by gianpaj** [Zhengqbbb/cz-git#153]
### 💭 Describe the feature

I think the default one is hardcoded to `gpt-3.5-turbo`
https://github.com/Zhengqbbb/cz-git/blob/b091c64cdfb3fa38d7ee4fdeefff1a03b024e4fb/packages/cz-git/src/generator/api.ts#L87

### 💡 Proposed Solution

Be able to set the AI model in the `czg` command line
e.g.
```
czg --ai-model=gpt-4-1106-preview
```

https://platform.openai.com/docs/models/gpt-4-and-gpt-4-turbo

**Comment by gianpaj** [comment]
Related but not the same as https://github.com/Zhengqbbb/cz-git/pull/102

As you can't specify the exact model. 

GPT-4 turbo (`gpt-4-1106-preview`) is 66% cheaper than GPT-4 (`gpt-4`)
<img width="490" alt="image" src="https://github.com/Zhengqbbb/cz-git/assets/899175/32c02e98-464c-4284-8e23-663ab4c2e157">

`gpt-3.5-turbo-1106` costs $0.0010 / 1K tokens

But `gpt-4-1106-preview` is 1000% more expensive than `gpt-3.5-turbo-1106` 😬💰 

https://openai.com/pricing

**Comment by Zhengqbbb** [comment]
Unfortunately, my openai api key has expired, and I can't renewal it, so the feature may not be landed in time.

Sure ~ PR Welcome 😉

**Comment by Zhengqbbb** [comment]
I have released a next pre-release version 🌟(**v1.10.0-beta.1**)🌟 for testing.
Plan for the official release on the weekend  🚀.

**Welcome to try it out! 👀 Let me know if there are still any issues. 🎉**

```sh
npm install -g czg@next
czg -v # ensure version is v1.10.0-beta.1
czg -h # show help

# Usage

### 1. Use the default `gpt-4o-mini` model
czg ai

### 2. Set specify a model in current session
czg ai -M=gpt-3.5-turbo

### 3. Setup the default model
czg --api-model="gpt-4"
```

See more setup details in https://deploy-preview-185--cz-git.netlify.app/recipes/openai#setup-openai-token

**Comment by Zhengqbbb** [comment]
The latest version 🌟(v1.10.0)🌟 should be resolve this issue request , please have a try. 💪
👀 Let me know if there are still any issues. 🎉
Plz check the document information: https://cz-git.qbb.sh/config/engineer#aimodel
